src/components/VideoCard/VideoCard.jsx

```diff
@@ -0,0 +1,47 @@
+import "./video-card.css";
+import { useState } from "react";
+import { PopUpMenu } from "../PopUpMenu/PopUpMenu";
+
+const VideoCard = ({ videoDetails }) => {
+  const { title, creator, views, thumbnail, img } = videoDetails;
+  const [showMenu, setShowMenu] = useState(false);
+  const titleNew = title.length > 36 ? title.slice(0, 33) + "..." : title;
```

Comment:
Nice idea to shorten the title....i had used css ellipsis property for this but this seems easier